workitem/link/type_repository.go

```diff
@@ -168,48 +169,84 @@ func (r *GormWorkItemLinkTypeRepository) Save(ctx context.Context, modelToSave W
+// ListSourceLinkTypes returns all the link types where the given work item type
+// can be used in the source of the link type. It does so by analysing the type
+// combinations that are associated with the given link type.
+func (r *GormWorkItemLinkTypeRepository) ListSourceLinkTypes(ctx context.Context, spaceID, witID uuid.UUID) ([]WorkItemLinkType, error) {
 	db := r.db.Model(WorkItemLinkType{})
-	query := fmt.Sprintf(`
+	where := fmt.Sprintf(`
 			-- Get link types we can use with a specific WIT if the WIT is at the
 			-- source of the link.
-			(SELECT path FROM %[2]s WHERE id = %[1]s.source_type_id LIMIT 1)
-			@>
-			(SELECT path FROM %[2]s WHERE id = ? LIMIT 1)`,
-		WorkItemLinkType{}.TableName(),
+			
+			-- filter by the current space and the system space
+			-- NOTE(kwk): Keep filtering by system space even with space templates!!!
+			space_id IN (?, ?)
+			
+			AND id IN (
+				SELECT link_type_id FROM %[1]s /* work_item_link_type_combinations */ AS combi
+				WHERE
+					-- filter by the current space and the system space
+					-- TODO(kwk): Not need to filter by system space when space templates are in
+					combi.space_id IN (?, ?)
+					AND
+					(SELECT path FROM %[2]s /* work_item_types */ WHERE id = combi.source_type_id LIMIT 1)
+					@>
+					(SELECT path FROM %[2]s /* work_item_types */ WHERE id = ? LIMIT 1)
+			)`,
+		WorkItemLinkTypeCombination{}.TableName(),
 		workitem.WorkItemType{}.TableName(),
 	)
-	db = db.Where(query, witID)
+	db = db.Where(where,
+		spaceID,
+		space.SystemSpace,
+		spaceID,
+		space.SystemSpace, /*TODO(kwk): not needed when space templates are in*/
+		witID)
 	var rows []WorkItemLinkType
 	db = db.Find(&rows)
 	if db.RecordNotFound() {
 		return nil, nil
 	}
-	if db.Error != nil {
-		return nil, errs.WithStack(db.Error)
-	}
-	return rows, nil
+	return rows, errs.WithStack(db.Error)
 }
 
-func (r *GormWorkItemLinkTypeRepository) ListTargetLinkTypes(ctx context.Context, witID uuid.UUID) ([]WorkItemLinkType, error) {
+// ListTargetLinkTypes returns all the link types where the given work item type
+// can be used in the target of the link type. It does so by analysing the type
+// combinations that are associated with the given link type.
+func (r *GormWorkItemLinkTypeRepository) ListTargetLinkTypes(ctx context.Context, spaceID, witID uuid.UUID) ([]WorkItemLinkType, error) {
 	db := r.db.Model(WorkItemLinkType{})
-	query := fmt.Sprintf(`
+	where := fmt.Sprintf(`
 			-- Get link types we can use with a specific WIT if the WIT is at the
 			-- target of the link.
-			(SELECT path FROM %[2]s WHERE id = %[1]s.target_type_id LIMIT 1)
-			@>
-			(SELECT path FROM %[2]s WHERE id = ? LIMIT 1)`,
-		WorkItemLinkType{}.TableName(),
+
+			-- filter by the current space and the system space
+			-- NOTE(kwk): Keep filtering by system space even with space templates!!!
+			space_id IN (?, ?)
+
+			AND id IN (
+				SELECT link_type_id FROM %[1]s /* work_item_link_type_combinations */ AS combi
```

Comment:
Ignore the `%[1]s` and just read `work_item_link_type_combinations`. That is how the comments are supposed to help me.